enhancements/vmaas/README.md

```diff
@@ -0,0 +1,472 @@
+- I want to manage (create, list, delete) my private network
+- I want to attach my VMs in a virtual private network
+- I want my VM to be exposed outside of my private network
+- I want my VMs to be attached to my own physical L2 network
+- I want to connect on my VM through serial console
+- I want to keep running my VM when a node goes out of service
+
+#### As a service provider:
+
+- I want to define VM templates
+- I want to manually trigger VM migration
+
+### Goals
+
+TBD
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+### Network isolation
+
+This design relies on UDN (User Defined Networking, it is the networking component provided with Openshift, it allows the definition of virtual private subnets, and provides the ability to connect on physical networks (localnet).
+
+Moverover, Kubevirt supports UDN layer 2 mode which offers the ability to live-migrate VMs across OCP nodes.
+
+### Ingress
+
+We can expose services running on VM(s) using a “LoadBalancer” service (e.g.: relying on MetalLB).
+
+### APIs
+
+Virtual private subnet creation:
+
+```
+apiVersion: cloudkit.openshift.io/v1alpha1
+kind: VirtualPrivateSubnet
+metadata:
+  name: example
+  finalizers:
+  - cloudkit.openshift.io/finalizer
+annotations:
+  cloudkit.openshift.io/ref-count: 1 # +1 when a VM is created, -1 when a VM is deleted  
+spec:
+  ipv4Cidr: 192.168.100.0/24
+```
+
+We will create a new, unique namespace, and the UDN definition to create a L2 layer and the corresponding subnet.
+
+Since several VMs can be attached to the same network, we introduce an annotation that will be used to count the number of VMs attached to this network, and a finalizer to prevent its deletion before all the VMs are deleted.
+
+Load balancer:
+
+```
+apiVersion: cloudkit.openshift.io/v1alpha1
+kind: LoadBalancer
+metadata:
+  name: example
+spec:
+  hostRefs:
+  - kind: VirtualMachineHost
+    name: exampleVM
+  portMapping:
+  - protocol: TCP
+    externalPort: 80
+    internalPort: 80
+```
+
+All hosts must be associated with the same network.
+
+### Template management
+
+Like cluster fulfillment, there will be a mechanism to define templates as Ansible roles. Templates contain predefined VM classes which define default OS base image, CPU architecture, number of cores, amount of memory, storage, user data, …
+
+A template VM might define the following properties, and let a tenant to customize some of them (should be a 1:1 mapping to KubeVirt resources):
+
+- Name
+- Tenant
+- OS base image
+- Instance class, which specify the family (translating into general/compute/memory/gpu/arch), and the amount of CPU and memory (TBD, e.g.: general.large)
+- User-data (cloud-init)
+- Networking, namespace + UDN at first, and localnet later?
+  * Layer 3 only for Networking.
+- SSH keys
+- Storage, size of the boot volume (+ I/O performance, encryption)
+- State (started/stopped)
+
+There will be an API to publish these templates in the fulfillment service, and list these templates from it.
```

Comment:
Presumably that API is not being created as part of this enhancement, right? It would be good to just add a couple of sentences then about how template distribution capabilities will be used here.